google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/ReadModifyWriteRow.java

```diff
@@ -0,0 +1,123 @@
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.data.v2.models;
+
+import com.google.api.core.InternalApi;
+import com.google.bigtable.v2.ReadModifyWriteRowRequest;
+import com.google.bigtable.v2.ReadModifyWriteRule;
+import com.google.bigtable.v2.TableName;
+import com.google.cloud.bigtable.data.v2.internal.RequestContext;
+import com.google.common.base.Preconditions;
+import com.google.protobuf.ByteString;
+import javax.annotation.Nonnull;
+
+/** Wraps a {@link ReadModifyWriteRowRequest}. */
+public final class ReadModifyWriteRow {
+  private final String tableId;
+  private final ReadModifyWriteRowRequest.Builder builder = ReadModifyWriteRowRequest.newBuilder();
+
+  private ReadModifyWriteRow(@Nonnull String tableId, @Nonnull ByteString key) {
+    Preconditions.checkNotNull(tableId, "tableId can't be null.");
+    Preconditions.checkNotNull(key, "key can't be null.");
+    this.tableId = tableId;
+
+    builder.setRowKey(key);
+  }
+
+  public static ReadModifyWriteRow create(@Nonnull String tableId, @Nonnull String key) {
+    return new ReadModifyWriteRow(tableId, ByteString.copyFromUtf8(key));
+  }
+
+  public static ReadModifyWriteRow create(@Nonnull String tableId, @Nonnull ByteString key) {
+    return new ReadModifyWriteRow(tableId, key);
+  }
+
+  /**
+   * Appends the value to the existing value of the cell. If the targeted cell is unset, it will be
+   * treated as containing the empty string.
+   */
+  public ReadModifyWriteRow append(
+      @Nonnull String familyName, @Nonnull String qualifier, @Nonnull String value) {
+    return append(familyName, ByteString.copyFromUtf8(qualifier), ByteString.copyFromUtf8(value));
+  }
+
+  /**
+   * Appends the value to the existing value of the cell. If the targeted cell is unset, it will be
+   * treated as containing the empty string.
+   */
+  public ReadModifyWriteRow append(
+      @Nonnull String familyName, @Nonnull ByteString qualifier, @Nonnull ByteString value) {
+    Validations.validateFamily(familyName);
+    Preconditions.checkNotNull(qualifier, "Qualifier can't be null");
+    Preconditions.checkNotNull(value, "Value can't be null");
+    Preconditions.checkArgument(!value.isEmpty(), "Value can't be empty");
+
+    ReadModifyWriteRule rule =
+        ReadModifyWriteRule.newBuilder()
+            .setFamilyName(familyName)
+            .setColumnQualifier(qualifier)
+            .setAppendValue(value)
+            .build();
+
+    builder.addRules(rule);
+
+    return this;
+  }
+
+  /**
+   * Adds `amount` be added to the existing value. If the targeted cell is unset, it will be treated
+   * as containing a zero. Otherwise, the targeted cell must contain an 8-byte value (interpreted as
+   * a 64-bit big-endian signed integer), or the entire request will fail.
+   */
+  public ReadModifyWriteRow increment(
+      @Nonnull String familyName, @Nonnull String qualifier, long amount) {
```

Comment:
Any limitation on 'amount'? can it be 0? negative?

Reply:
I don't know of any limitations, but I think those validations are best left to the server